### ISSUE TITLE
runScript specification 

### ISSUE THREAD

**Issue by flowernert** [ninia/jep#166]
- OS Windows 10 Pro
- Python 3.6 Anaconda
- Java 8 Oracle
- Jep 3.8.2
- Python packages: pyplot

The specification of the Jep.runScript(script) doesn't specify if it will run the __main__ of a python script.

It's likely that it doesn't during my experiments, but I'm not totally sure.

The documentation should specify this point, I thought I had a bug in my code and spent quite some time on it, and it was just that the main of the script was not executed, maybe we can avoid this to other people 😄 .

I tried to figure out by myself looking at the C code but I'm lost at the PyRun_File function leading me to nowhere.

I designed this experiment to figure out, call the following script from Java `Jep.runscript(script)`, then call `jep.getValue(val)`. The value got back is 0.

```
val = 0

def change_val():
    global val
    val = 42

if __name__ == "__main__":
    change_val()
```

**Comment by bsteffensmeier** [comment]
I just ran across this [here](https://github.com/ninia/jep/commit/93c1918a2663f6318b4ed064ff1a294e7f65669a#diff-b9918ab2ccfbd2412725c6fd755d85cfR64). I didn't realize there was already an issue for the problem. I worked around it by manually specifying the `__name__`, which should work for anyone with problems on the current release.

In my testing `__name__` is `'__main__'` on a regular Jep instance but on a SharedInterpreter  `__name__` is `'__builtin__'` os `builtins` depending on Python 2 or 3. This is consistent if you are using runScript() or even just an eval(). We want SharedInterpreter to be a drop-in replacement for Jep so I think we should just set  `__name__` to `'__main__'` in the constructor for SharedInterpreter.

**Comment by bsteffensmeier** [comment]
This should be resolved with 1756053 in the dev_3.9 branch

**Comment by bsteffensmeier** [comment]
Jep 3.9.0 has been released
